### ISSUE TITLE
Make port publishing easier

### ISSUE THREAD

**Issue by larsks** [CCI-MOC/esi#678]
Port forwarding is a nice way to expose services, but the process is too complicated. Let's say that I want to "forward tcp port 4321 from a public ip address to port 4321 on internal address 10.100.100.10". That means:

1. Find the existing port with 10.100.100.10, or create a new port with that address if one doesn't already exist.
2. Create a floating ip address.
3. Take the information from the first two steps and plug it into an `openstack floating ip publishing` command

It would be really neat if instead I could write:

```
openstack esi publish --network mynetwork --from 10.100.100.10:4321
```

And have it perform all the above steps. If I want to use a specific floating ip:

```
openstack esi publish --network mynetwork --from 10.100.100.10:4321 --to 128.31.20.112
```

And if I want the external port to be different from the internal port:

```
openstack esi publish --network mynetwork --from 10.100.100.10:4321 --to 128.31.20.112:1234
```

(I've gone with the verb `publish` here, because that's what we use in the container world, but we could just as easily use something like `port forward`)

**Comment by tzumainn** [comment]
Fair enough. I'd probably go for the "port forwarding" syntax just because it matches existing OpenStack commands better.

**Comment by larsks** [comment]
E.g. see the portforward.sh script [here](https://github.com/larsks/esi-tools/tree/main).
